code/seg.c

```diff
@@ -292,25 +295,35 @@ void SegSetRankSet(Seg seg, RankSet rankSet)
 {
   AVERT(Seg, seg);
   AVERT(RankSet, rankSet);
-  AVER(rankSet != RankSetEMPTY || SegSummary(seg) == RefSetEMPTY);
+  /* FIXME: Hidden cast to GCSeg in the next line */
+  AVER(rankSet != RankSetEMPTY || RefSetIsEmpty(segSummary(seg)));
   Method(Seg, seg, setRankSet)(seg, rankSet);
 }
 
 
+/* SegGetSummary -- get the summary from a segment */
+
+void SegGetSummary(RefSet summaryReturn, Seg seg)
+{
+  AVERT(Seg, seg);
+  RefSetCopy(summaryReturn, segSummary(seg));
+}
+
+
 /* SegSetSummary -- change the summary on a segment */
 
 void SegSetSummary(Seg seg, RefSet summary)
 {
   AVERT(Seg, seg);
-  AVER(summary == RefSetEMPTY || SegRankSet(seg) != RankSetEMPTY);
+  AVER(RefSetIsEmpty(summary) || SegRankSet(seg) != RankSetEMPTY);
 
 #if defined(REMEMBERED_SET_NONE)
   /* Without protection, we can't maintain the remembered set because
      there are writes we don't know about. */
-  summary = RefSetUNIV;
+  summary = RefSetUniv;
```

Comment:
This should be a RefSetCopy, not an assignment.

Reply:
The reason is that this is a structure assignment.  The MPS forbids them because they can cause an implicit dependency on memcpy, and the MPS is freestanding (does not depend on the C library, ref ISO C).

We should have an automated check that the MPS is freestanding.

Reply:
Fixed in fda95fa1f